airflow/contrib/operators/gcp_container_operator.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+from airflow import AirflowException
+from airflow.contrib.hooks.gcp_container_hook import GKEClusterHook
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class GKEClusterDeleteOperator(BaseOperator):
+    template_fields = ['project_id', 'gcp_conn_id', 'name', 'location', 'api_version']
+
+    @apply_defaults
+    def __init__(self,
+                 project_id,
+                 name,
+                 location,
+                 gcp_conn_id='google_cloud_default',
+                 api_version='v2',
+                 *args,
+                 **kwargs):
+        """
+
+        :param project_id: The Google Developers Console [project ID or project number]
+        :type project_id: str
+        :param name: The name of the resource to delete, in this case cluster name
+        :type name: str
+        :param location: The name of the Google Compute Engine zone in which the cluster
+        resides.
+        :type location: str
+        :param gcp_conn_id: The connection ID to use connecting to Google Cloud Platform.
+        :type gcp_conn_id: str
+        :param api_version: The api version to use
+        :type api_version: str
+        """
+        super(GKEClusterDeleteOperator, self).__init__(*args, **kwargs)
+
+        self.project_id = project_id
+        self.gcp_conn_id = gcp_conn_id
+        self.location = location
+        self.api_version = api_version
+        self.cluster_name = name
+
+    def _check_input(self):
+        if not self.project_id:
+            self.log.error('Please specify the project_id of cluster to use.')
+            raise AirflowException('project_id parameter missing, unable to trigger job')
+        if not self.cluster_name:
+            self.log.error('Please specify the cluster\'s name to delete.')
+            raise AirflowException('cluster_name parameter missing, unable to start job')
+        if not self.location:
+            self.log.error('Please specify the cluster location id to use.')
+            raise AirflowException('location parameter missing, unable to trigger job')
```

Comment:
nit: if not all([self.project_id, self.name, self.location]):
  self.log.error(...)
   raise AirflowException...